blog/2023-05-03-understanding-entropy-the-key-to-secure-cryptography-and-randomness/index.md

```diff
@@ -0,0 +1,141 @@
+
+## Which applications usually use entropy?
+
+There are many applications that rely on entropy to generate random numbers, keys, or other cryptographic elements.
+
+### Cryptographic protocols
+
+Cryptographic protocols, such as SSL/TLS, SSH, and IPsec rely heavily on entropy to generate random numbers and keys for encryption, authentication, and integrity protection.
+
+The randomness of these numbers is crucial for the security of the communication. If an attacker can predict the random numbers used in a cryptographic protocol, they can potentially compromise the confidentiality, integrity, or authenticity of the communication.
+
+For example, in SSL/TLS, the server generates a random number called the "pre-master secret" and sends it to the client encrypted with the public key of the server's certificate. The client then decrypts the pre-master secret with its own private key and uses it to generate the "master secret," which is used to derive the session keys for encryption and authentication.
+
+If the pre-master secret is predictable or biased, an attacker can potentially intercept the encrypted message, decrypt it, and gain access to the session keys. This would allow the attacker to eavesdrop on the communication, modify the data, or impersonate the server.
+
+To prevent this type of attack, SSL/TLS requires a sufficient amount of high-quality entropy to generate the pre-master secret. The server generates the pre-master secret by combining random data from various sources, such as hardware interrupts, keyboard presses, and mouse movements. This ensures that the pre-master secret is unpredictable and random.
+
+Similarly, in SSH, the client and server generate random numbers to establish a shared secret key for encryption and authentication. If the random numbers are predictable or biased, an attacker can potentially intercept the communication and compromise its security.
+
+To prevent this type of attack, SSH requires a sufficient amount of high-quality entropy to generate the random numbers. The client and server generate the random numbers by combining random data from various sources, such as hardware interrupts, keyboard presses, and mouse movements. This ensures that the random numbers are unpredictable and random.
+
+### Database servers
+
+Database servers can use entropy in various ways, depending on the specific database system and its configuration. Here are some examples:
+
+- **Encryption:** Many database systems support encryption of data-at-rest or data-in-transit, which relies on random numbers for generating encryption keys, initialization vectors, and other cryptographic elements. For example, Microsoft SQL Server provides Transparent Data Encryption (TDE), which encrypts the data stored on disk, and uses a random number generator to create the encryption key.
```

Comment:
I thought we're adding the bold before the ":". Minor, as I didn't notice it at all in the HTML anyway.